File: clients/java/src/main/java/model/Player.java
package model;

import util.StreamUtil;

public class Player {
    private int id;
    public int getId() { return id; }
    public void setId(int id) { this.id = id; }
    private int score;
    public int getScore() { return score; }
    public void setScore(int score) { this.score = score; }
    public Player() {}
    public Player(int id, int score) {
        this.id = id;
        this.score = score;
    }
    public static Player readFrom(java.io.InputStream stream) throws java.io.IOException {
        Player result = new Player();
        result.id = StreamUtil.readInt(stream);
        result.score = StreamUtil.readInt(stream);
        return result;
    }
    public void writeTo(java.io.OutputStream stream) throws java.io.IOException {
        StreamUtil.writeInt(stream, id);
        StreamUtil.writeInt(stream, score);
    }
}

File: clients/javascript/model/player.js
class Player {
    constructor(id, score) {
        this.id = id;
        this.score = score;
    }
    static async readFrom(stream) {
        let id;
        id = await stream.readInt();
        let score;
        score = await stream.readInt();
        return new Player(id, score);
    }
    async writeTo(stream) {
        let id = this.id;
        await stream.writeInt(id);
        let score = this.score;
        await stream.writeInt(score);
    }
}
module.exports = { Player: Player }
